Sources/NIOWebSocket/WebSocketUpgrader.swift

```diff
@@ -0,0 +1,126 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebSocketUpgrader.swift
+//  NIOWebSocket
+//
+//  Created by Cory Benfield on 14/02/2018.
+//
+
+import class Foundation.NSNull // for string operations
```

Comment:
This reads like magic... I guess it's for `trimmingCharacters()` but that method is on `NSString` not `NSNull`? Confused.

Reply:
A small number of `String` methods from the stdlib are implemented by delegating to their `Foundation` counterparts on `NSString`. These methods don't work if you don't force the compiler to link `Foundation`: you'll get linker errors because the symbols don't resolve.

This is our way of saying that we don't actually need any object from `Foundation` in our namespace: we just need the side-effect of getting the linker to link the library.

Reply:
@ianpartridge there's a weird thing in Swift: As soon as you import _anything_ from `Foundation`, you'll get all the `extension`s defined on `NSString` available on `Swift.String`. That's only true for `String` and super weird but I guess a stop gap until `Swift.String` itself has more than a handful of methods.

Reply:
Thanks, both of you. It is indeed very weird!